src/ast/ExecutionContext.ts

```diff
@@ -3,6 +3,7 @@ import { PathTracker } from './utils/PathTracker';
 import ThisVariable from './variables/ThisVariable';
 
 interface ExecutionContextIgnore {
+	breakAndContinue: boolean;
 	breakStatements: boolean;
```

Comment:
Assuming this isn't a public interface, rather than having the overlap of concerns between `breakAndContinue` and `breakStatements`, I'd suggest to simplify this and have distinct `breakStatements` and `continueStatements`. Yes, the meaning of the terms would change and would require some more refactoring, but there'd be less cognitive overload.

Reply:
I was considering this and expect this version to be slightly more performant than the alternative in real world scenarios (loops without breaks are more common than loops with breaks and that way, you only need to check both flags for break statements while otherwise, you need to set and reset both flags for each kind of loop, which also results in more code) but I guess you are right that the advantage is negligible.